libcontainer/cgroups/fs/memory_test.go

```diff
@@ -492,3 +497,49 @@ func TestMemorySetOomControl(t *testing.T) {
 		t.Fatalf("Got the wrong value, set memory.oom_control failed.")
 	}
 }
+
+func TestMemorySetMemorNegativeToReset(t *testing.T) {
+	helper := NewCgroupTestUtil("memory", t)
+	defer helper.cleanup()
+
+	const (
+		memoryBefore     = 314572800 // 300M
+		memoryswapBefore = 629145600 // 600M
+		memoryAfter      = 0         // Unlimited
```

Comment:
Why 0 presents unlimited? Doesn't sound reasonable. Note that unit test didn't use real cgroup fs for test, so you better add an integration test to cover this case.